docs/KubeflowMode.md

```diff
@@ -1,115 +1,115 @@
+```
+kubeflowConfig:
+  operator: tf-operator
+  storage: azureStorage
+  keyVault:
+    vaultName: {your_vault_name}
+    name: {your_secert_name}
+  azureStorage:
+    accountName: {your_storage_account_name}
+    azureShare: {your_azure_share_name}
+```
+
+Note: You should explicitly set `trainingServicePlatform: kubeflow` in nni config yaml file if you want to start experiment in kubeflow mode. 
+
+Trial configuration in kubeflow mode have the following configuration keys:
+* codeDir
+    * code directory, where you put training code and config files
+* worker (required). This config section is used to configure tensorflow worker role
+    * replicas
+        * Required key. Should be positive number depends on how many replication your want to run for tensorflow worker role.
+    * command
+        * Required key. Command to launch your trial job, like ```python mnist.py```
+    * memoryMB
+        * Required key. Should be positive number based on your trial program's memory requirement
+    * cpuNum
+    * gpuNum
+    * image
+        * Required key. In kubeflow mode, your trial program will be scheduled by Kubernetes to run in [Pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/). This key is used to specify the Docker image used to create the pod where your trail program will run. 
+        * We already build a docker image [nnimsra/nni](https://hub.docker.com/r/msranni/nni/) on [Docker Hub](https://hub.docker.com/). It contains NNI python packages, Node modules and javascript artifact files required to start experiment, and all of NNI dependencies. The docker file used to build this image can be found at [here](../deployment/Dockerfile.build.base). You can either use this image directly in your config file, or build your own image based on it.
+* ps (optional). This config section is used to configure tensorflow parameter server role.
+
+Once complete to fill nni experiment config file and save (for example, save as exp_kubeflow.yaml), then run the following command
+```
+nnictl create --config exp_kubeflow.yaml
+```
+to start the experiment in kubeflow mode. NNI will create Kubeflow tfjob for each trial, and the job name format is something like `nni_exp_{experiment_id}_trial_{trial_id}`. 
+You can see the kubeflow tfjob created by NNI in your Kubernetes dashboard. 
+
+Notice: In kubeflow mode, NNIManager will start a rest server and listen on a port which is your NNI WebUI's port plus 1. For example, if your WebUI port is `8080`, the rest server will listen on `8081`, to receive metrics from trial job running in Kubernetes. So you should `enable 8081` TCP port in your firewall rule to allow incoming traffic. 
+
+Once a trial job is completed, you can goto NNI WebUI's overview page (like http://localhost:8080/oview) to check trial's information. 
+
+Any problems when using NNI in kubeflow mode, plesae create issues on [NNI github repo](https://github.com/Microsoft/nni), or send mail to nni@microsoft.com
+
```

Comment:
why modify kubeflow doc?

Reply:
Yes, I merge the master branch every time and keep the incoming change